### DISCUSSION TITLE
Self-hosting: what are the minimal required resources?

### DISCUSSION THREAD

**Discussion by lemeurherve** [badges/shields#8187] [Q&A] [answered]
As [we want to self-host an instance for ci.jenkins.io](https://github.com/jenkins-infra/helpdesk/issues/3044), we're wondering what are the minimal required resources (CPU/mem)?

I might have missed it but I didn't found this info in https://github.com/badges/shields/blob/master/doc/releases.md#shields-server or https://github.com/badges/shields/blob/master/doc/self-hosting.md

**Comment by calebcartwright** [comment, marked as answer]
I'm not sure if we've ever pushed to the extreme of trying to find out the absolute bare minimum for the server to be able to spin up. In general, you'll find it's possible to run the server on the smallest available offering of PaaS and SaaS type platforms.

Your mileage may vary, but 1 vCPU (or even fractional CPU for those platforms that do so) and 256 MB of memory should provide plenty of leg room. I should also add that we've typically observed rapidly diminishing returns from vertical scaling, and rely exclusively on horizontal scaling to handle the load for Shields.io
